tests/integration/conftest.py

```diff
@@ -142,40 +79,90 @@ def fixture_kafka_server(request, session_tmppath: Path) -> Iterator[KafkaServer
         yield kafka_servers
         return
 
-    kafka_dir = session_tmppath / "kafka"
-    zk_dir = session_tmppath / "zk"
-    transfer_file = session_tmppath / "zk_kafka_config"
+    zk_dir = session_logdir / "zk"
 
-    with ExitStack() as stack:
-        # Synchronize xdist workers, data generated by the winner is shared through
-        # transfer_file (primarily the server's port number)
+    # File used to share data among test runners, including the dynamic
+    # configuration for this session (mainly port numbers), and and
+    # synchronization data.
+    transfer_file = session_logdir / "transfer"
+    lock_file = lock_path_for(transfer_file)
```

Comment:
This is why the `session_logdir` needs to be a shared dir, so that each work uses the same file as a lock.